### DISCUSSION TITLE
Eclipse Mosquitto (MQTT LXC) cannot be ascessed from network

### DISCUSSION THREAD

**Discussion by pedjas** [tteck/Proxmox#1671] [Help Needed]
I installed Mosquitto using script, and set it up as instructed. It is not accessible from network. It is like port is closed or blocked.

When I try to connect using telnet to external IP and port 1883 from within console on Mosquito LXC - that works. Telnet connects. So it means Mosquitto is up and running and it listens on external IP.

Before this, I installed few other LXC using scripts (nginx and pihole) and they work fine. I checked all I could in Proxmox and all configuration is the same for all three LXCs. Firewall is off so I guess it is not Proxmox that blocks access to port.

### /etc/mosquitto/conf.d/default.conf:
`allow_anonymous false

persistence true

password_file /etc/mosquitto/passwd

listener 1883`

I tried setting up IP for listener, but it did not make any change.

### LXC config:
`## MQTT LXC

arch: amd64

cores: 1

features: keyctl=1,nesting=1

hostname: mqtt

memory: 512

net0: name=eth0,bridge=vmbr0,hwaddr=AA:B3:58:FB:BC:B3,ip=dhcp,type=veth

onboot: 1

ostype: debian

rootfs: local-lvm:vm-105-disk-0,size=2G

swap: 512

tags: proxmox-helper-scripts

unprivileged: 1`

Proxmox is version 8.0.3

Any suggestion what should I check for?

**Comment by tteck** [comment]
When connecting to a MQTT broker, credentials are required.

**Reply by pedjas** [reply]
That is not an issue. When I try to connect using telnet from within VM it connects, but as telnet does not follow protocol it cannot do more. But it connects.

When I try to connect from the network, it cannot establish connection at all, neither telnet not mqtt clients (with credentials). It is like port is closed, not that login is refused. It does not come to login.

**Reply by tteck** [reply]
What is the output of the command `ss -tul`?

**Reply by pedjas** [reply]
```
root@mqtt:~# ss -tul
Netid         State          Recv-Q         Send-Q                   Local Address:Port                     Peer Address:Port         Process         
udp           UNCONN         0              0                              0.0.0.0:bootpc                        0.0.0.0:*                            
tcp           LISTEN         0              100                            0.0.0.0:1883                          0.0.0.0:*                            
tcp           LISTEN         0              100                          127.0.0.1:smtp                          0.0.0.0:*                            
tcp           LISTEN         0              4096                                 *:ssh                                 *:*                            
tcp           LISTEN         0              100                               [::]:1883                             [::]:*                            
tcp           LISTEN         0              100                              [::1]:smtp                             [::]
```

**Reply by tteck** [reply]
Shows listening on port 1883  🤷🏻‍♂️

**Reply by pedjas** [reply]
Yes. As I said, when I am in MQTT server console, it seems to work. But when i try to connect from the network it does not.

SSHD is accessible from outside, for some reason ti does not allow root to log in (although I set sshd_config to allow it). But it does connect from outside.

Quite strange. Especially that previous two LXC scripts set everything flawlessly.

**Comment by pedjas** [comment]
This became even more strange.

I cannot connect to mqtt from my desktop PC, nor from HomeAssistant that is in separate VM but on the same Proxmox as mqtt.

But, I tried from another PC in same network and it connects and logs in mqtt flawlessly.

**Reply by tteck** [reply]
Have you tried deleting the existing MQTT LXC and then creating a new one?
It appears that you could be encountering an issue with your network.

**Reply by pedjas** [reply]
That is only one left to try. I doubt it is network related as everything is on one simple switch. Simple home network.

**Comment by pedjas** [comment]
Hm, I restarted everything in network and that resolved an issue. Mqtt became accessible as expected.
